packages/go_router/test/go_router_delegate_test.dart

```diff
@@ -23,36 +23,46 @@ GoRouterDelegate createGoRouterDelegate({
     ],
     refreshListenable: refreshListenable,
   );
-  return router.routerDelegate;
+  await tester.pumpWidget(MaterialApp.router(
+      routeInformationProvider: router.routeInformationProvider,
+      routeInformationParser: router.routeInformationParser,
+      routerDelegate: router.routerDelegate));
+  return router;
 }
 
 void main() {
   group('pop', () {
-    test('removes the last element', () {
-      final GoRouterDelegate delegate = createGoRouterDelegate()
-        ..push('/error')
-        ..addListener(expectAsync0(() {}));
-      final GoRouteMatch last = delegate.matches.last;
-      delegate.pop();
-      expect(delegate.matches.length, 1);
-      expect(delegate.matches.contains(last), false);
+    testWidgets('removes the last element', (WidgetTester tester) async {
+      final GoRouter goRouter = await createGoRouter(tester)
+        ..push('/error');
+
+      goRouter.routerDelegate.addListener(expectAsync0(() {}));
+      final GoRouteMatch last = goRouter.routerDelegate.matches.last;
+      goRouter.routerDelegate.pop();
+      expect(goRouter.routerDelegate.matches.length, 1);
+      expect(goRouter.routerDelegate.matches.contains(last), false);
```

Comment:
If a developer tests their app that uses `go_router`, will the breaking changes in 4.0.0 break their tests? If yes, should that be covered in the breaking changes doc? flutter.dev/go/go-router-v4-breaking-changes

Reply:
A test that uses the GoRouter API? Yeah upgrading to `4.0.0` will probably break those tests. We could warn users about that, but we aren't already documenting the underlying reasons in the doc?

Reply:
I asked as the changes in this test file aren't covered by the migration doc: flutter.dev/go/go-router-v4-breaking-changes

I'd do a spot check on the migration doc and make sure there's no low hanging fruits related to testing there. If you feel the migration doc is good enough already, feel free to resolve this comment 😄